release.nix

```diff
@@ -244,49 +122,25 @@ let
                 };
 
               in
-                composeExtensionList pkgsNew.lib
-                  [ upgradeOverrides
-                    patchOverrides
-                    dontCheckOverrides
-                    jailbreakOverrides
+                pkgsNew.lib.foldr pkgsNew.lib.composeExtensions (old.overrides or (_: _: { }))
+                  [ directoryOverrides
                     manualOverrides
                   ];
-          };
+          });
         };
       };
-    })
-  ];
-
-  unpatchedPkgs = import unpatchedNixpkgs { inherit config overlays; };
-
-  # https://github.com/NixOS/nixpkgs/pull/85446
-  nixpkgs = unpatchedPkgs.stdenvNoCC.mkDerivation {
-    name = "nixpkgs-${nixpkgsRelease}-patched";
+    };
 
-    src = unpatchedNixpkgs;
+  overlays = [ overlay ];
 
-    # Backport fix <https://github.com/NixOS/nixpkgs/pull/85446> to 20.03:
-    patches = [ ./nix/with-packages-wrapper.patch ];
-
-    phases = [ "unpackPhase" "patchPhase" "installPhase" ];
-
-    installPhase = ''
-      mkdir -p $out
-      cp -R ./ $out/
-    '';
-
-    preferLocalBuild = true;
-  };
+  pkgs = import nixpkgs { inherit config overlays; };
 
   linuxPkgs =
     import nixpkgs { inherit config overlays; system = "x86_64-linux" ; };
 
   darwinPkgs =
     import nixpkgs { inherit config overlays; system = "x86_64-darwin"; };
```

Comment:
Is it really necessary for us to create system-specific attributes, when the user could just specify `--system` at the command-line themselves?